### PR TITLE
Document value-set analysis

### PR DESCRIPTION
**Pull request by owen-mc-diffblue**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/pointer-analysis/value_set_domain.h [hunk 1]
```diff
@@ -17,6 +17,9 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include "value_set.h"
 
+/// This is the domain for a value set analysis. It is intended to be the
+/// template parameter for`value_set_analysis_templatet`, so `VST` would be
```
Comment (peterschrammel) [inline src/pointer-analysis/value_set_domain.h]: space missing after `for`

**Review by peterschrammel — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 38b0dff0466a98cb38eafdd78996ba17e4a960c2).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99810347
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

File: src/pointer-analysis/value_set_analysis.h [hunk 2]
```diff
@@ -26,6 +26,13 @@ void value_sets_to_xml(
   const goto_programt &goto_program,
   xmlt &dest);
 
+/// This template class implements a static analysis which keeps track of what
```
Comment (tautschnig) [inline src/pointer-analysis/value_set_analysis.h]: I'd use "data-flow analysis" instead of "static analysis"

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 379357a600e4f6ddba75ea18eb4fd3a314536003).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99820906

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 04ac4259cdcb89691d969eb646a3615bf5e39200).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99912854

**Pull request merged**